hbase-client/src/main/java/org/apache/hadoop/hbase/client/SimpleRequestController.java

```diff
@@ -328,6 +346,16 @@ public void waitForFreeSlot(long id, int periodToTrigger, Consumer<Long> trigger
     waitForMaximumCurrentTasks(maxTotalConcurrentTasks - 1, id, periodToTrigger, trigger);
   }
 
+  @Override
+  public void waitForFreeSlot(int numberOfTask,long id, int periodToTrigger, Consumer<Long> trigger) throws InterruptedIOException {
+    waitForMaximumCurrentTasks(maxTotalConcurrentTasks - numberOfTask, id, periodToTrigger, trigger);
+  }
+
+  @Override
+  public void waitForAllFreeSlot(long id) throws InterruptedIOException {
+    waitForMaximumCurrentTasks(0, id, 1000000, null);
```

Comment:
What does the 1000000 mean?

Reply:
I change it to 0. This is unused as the trigger is null.